samples/subsys/usb/hid-keyboard/src/main.c

```diff
@@ -279,6 +279,17 @@ int main(void)
 			continue;
 		}
 
+		if (usbd_is_suspended(sample_usbd)) {
+			/* on a press of any button, send wakeup request */
+			if (kb_evt.value) {
+				ret = usbd_wakeup_request(sample_usbd);
+				if (ret) {
+					LOG_ERR("Remote wakeup error, %d", ret);
+				}
+			}
+			continue; /* cannot communicate in suspend */
+		}
+
```

Comment:
I do not think it should be handled by the application but by the stack (https://github.com/zephyrproject-rtos/zephyr/pull/73362).

Reply:
I will add my two cents on that MR there, here I can only propose:
We could still merge this for the time window between now and when that MR is finally ready, at which point these changes can be undone. I'd really like to have a simple reference sample for evaluating this functionality sooner rather than later.

Reply:
> We could still merge this for the time window between now and when that MR is finally ready, at which point these changes can be undone.I'd really like to have a simple reference sample for evaluating this functionality sooner rather than later.

Why do you need a "simple reference sample" in the tree to evaluate it? Why can you not use shell command for evaluation?

Reply:
Shell command is not quite suitable for USB certification, while a simple button press in hid-keyboard sample is.

Reply:
> Shell command is not quite suitable for USB certification, while a simple button press in hid-keyboard sample is.

This I fully agree with. It's valuable to have a single step way of evaluating the remote wakeup functionality, if there's a sample that is easily distributed, and the end user just needs to select the target board, build and flash as-is. Remote wakeup is a pretty widely used functionality for HID keyboards, so from that perspective I'd also rather have this than not.

I have no idea how the shell command functionality would work, but I have to clarify here that remote wakeup has to be evaluateable as part of a whole USB operation. (We found an issue in our design that the USB HID interface isn't responsive after remote wakeup, which was only observed with NCS 2.6.1, but not with NCS 2.7.0. Not exactly relevant here, I'm just trying to make a point.)